utils/src/main/scala/za/co/absa/enceladus/utils/fs/FileSystemUtils.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package za.co.absa.enceladus.utils.fs
+
+import java.net.URI
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.FileSystem
+import org.slf4j.{Logger, LoggerFactory}
+import za.co.absa.atum.utils.S3Utils.StringS3LocationExt
+
+import scala.collection.concurrent.TrieMap
+
+object FileSystemUtils {
+
+  val log: Logger = LoggerFactory.getLogger(this.getClass)
+
+  /**
+   * Will yeild a [[FileSystem]] for path. If path prefix suggest S3, S3 FS is returned, HDFS otherwise.
+   * @param path
```

Comment:
Missing descriptions for parameters - I wonder if it's better to have them empty or not at all.

Reply:
Thanks, desc added.